odoo/tools/misc.py

```diff
@@ -1380,6 +1380,26 @@ def _format_time_ago(env, time_delta, lang_code=False, add_direction=True):
     return babel.dates.format_timedelta(-time_delta, add_direction=add_direction, locale=locale)
 
 
+def humanize_number(number, decimal=2):
+    for unit in ['', 'k', 'M', 'G']:
+        if abs(number) < 1000.0:
+            return f"%3.{decimal}f%s" % (number, unit)
+        number /= 1000.0
+    return f"%.{decimal}f%s" % (number, 'T')
+
+
+def format_human_readable_amount(amount, currency=None):
+    formated_amount = humanize_number(amount)
+
+    if not currency:
+        return formated_amount
+
+    elif currency.position == 'before':
+        return "%s %s" % (currency.symbol or '', formated_amount)
```

Comment:
```
>>> format_human_readable_amount(9999, usd)                                                                                                                                                                                               
'$ 10.00k'
```
The currency position and presence or not of a space is actually defined on the language and not the currency (e.g. [en_US locale definition](https://lh.2xlibre.net/locale/en_US/)). Moving the currency position on the language is out of the scope of this PR but if we assume en_US as a standard ('Merica!), there is no space for currencies preceeding the amount.

Also, I find it a bit odd the default of 2 decimals. If I want to display a short version of a number, I don't want much precision. I will need the exact number for payment but if the point is to get a rough idea, one decimal is probably enough. But this is more subjective.

Reply:
@mart-e But there's a field `position` on the model currency...  So it this PR, I just remove the space ? (and change to one decimal by default :p )

Reply:
@std-odoo yes in this PR, just remove the space for the currencies with position `before`. For the position `after`, I am not sure, if we take fr_BE as a good reference ('elgica!), there is a space between the currency and amount. But then it's a short version... I guess it depends where it's displayed if it looks weird or not ^^